test/performance/shmem_perf_suite/bw_common.h

```diff
@@ -541,10 +547,34 @@ void static inline calc_and_print_results(double total_t, int len,
     /* base case: will be overwritten by collective if num_pes > 2 */
     pe_bw_sum = bw;
 
-    if(nPEs >= 2)
-        shmem_double_sum_to_all(&pe_bw_sum, &bw, nred_elements, start_pe,
+    if (metric_info.individual_report == 1) {
+        printf("Individual bandwith for PE %d is %10.2f\n", metric_info.my_node, pe_bw_sum);
+    }
+    
+    pe_time = total_t;
+    shmem_barrier(start_pe, stride, nPEs, red_psync);
+    if (nPEs >= 2) {
+        shmem_double_max_to_all(&pe_time_max, &pe_time, nred_elements, start_pe,
                                 stride, nPEs, pwrk,
                                 red_psync);
+    }
+
+    /* calculating bandwidth based on the highest time taken across all PEs */
+    if (pe_time_max > 0 ) {
```

Comment:
What should we do if `pe_time_max` is not greater than 0?

Reply:
`bw` is already assigned 0, so in terms of calculation, there is nothing to be done. But, we can definitely add an error message like the impossible case has happened.